File: src/main/java/org/smart4j/framework/controller/CustomerController.java
package org.smart4j.framework.controller;

import org.smart4j.framework.annotation.Action;
import org.smart4j.framework.annotation.Controller;
import org.smart4j.framework.annotation.Inject;
import org.smart4j.framework.model.Customer;
import org.smart4j.framework.service.CustomerService;

import java.util.List;
import java.util.Map;

/**
 * Created by zk on 01/11/2017.
 */
@Controller
public class CustomerController {

    @Inject
    CustomerService customerService;


//    @Override
//    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
//        List<Customer> customerList = customerService.getCustomerList();
//        req.setAttribute("customerList",customerList);
//        req.getRequestDispatcher("/WEB-INF/view/customer.jsp").forward(req,resp);
//    }
@Action("get:/")
public View index(Param param){
    List<Customer> customerList = customerService.getCustomerList();
    return new View("customer.jsp").addModel("customerList",customerList);
}

    @Action("get:/customer")
    public View getindex(Param param){
        List<Customer> customerList = customerService.getCustomerList();
        return new View("customer.jsp").addModel("customerList",customerList);
    }
    @Action("get:/customer_json")
    public Data getindex2(Param param){
        List<Customer> customerList = customerService.getCustomerList();
        return new Data(customerList);
    }

    @Action("get:/customer_create")
    public View create(Param param){
        return new  View("customer_create.jsp");
    }

    @Action("post:/customer_create")
    public Data createSubmit(Param param){
        Map<String,Object> fields=param.getMap();
        boolean customer = customerService.createCustomer(fields);
        return new Data(customer);
    }

    @Action("get:/customer_edit")
    public View edit(Param param){
        Long id =param.getLong("id");
        Customer customer = customerService.getCustomer(id);
        return new  View("customer_edit.jsp").addModel("customer",customer);
    }

    @Action("put:/customer_edit")
    public Data editSubmit(Param param){
        Long id =param.getLong("id");
        Map<String,Object> fields=param.getMap();
        boolean customer = customerService.updateCustomer(id,fields);
        return new Data(customer);
    }

    @Action("get:/customer_delete")
    public View delete(Param param){
        Long id =param.getLong("id");
        Boolean aBoolean = customerService.deleteCustomer(id);

        List<Customer> customerList = customerService.getCustomerList();
        return new View("customer.jsp").addModel("customerList",customerList);
    }
}

File: src/main/java/com/zk/controller/CustomerController.kt
package com.zk.controller

import com.zk.service.CustomerService
import org.summer.framework.annotation.Action
import org.summer.framework.annotation.Controller
import org.summer.framework.annotation.Inject
import org.summer.framework.controller.Data
import org.summer.framework.controller.Param
import org.summer.framework.controller.View
import java.util.*

/**
 * Created by zk on 01/11/2017.
 */
@Controller
class CustomerController {

    @Inject
    internal var customerService: CustomerService? = null


    //    @Override
    //    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
    //        List<Customer> customerList = customerService.getCustomerList();
    //        req.setAttribute("customerList",customerList);
    //        req.getRequestDispatcher("/WEB-INF/view/customer.jsp").forward(req,resp);
    //    }

    @Action("post:/")
    fun post(param: Param): View {
        val customerList = customerService!!.customerList
        return View("customer.jsp").addModel("customerList", customerList)
    }

    @Action("get:/")
    fun index(param: Param): View {
        val customerList = customerService!!.customerList
        return View("customer.jsp").addModel("customerList", customerList)
    }

    @Action("get:/customer")
    fun getindex(param: Param): View {
        val customerList = customerService!!.customerList
        return View("customer.jsp").addModel("customerList", customerList)
    }

    @Action("get:/customer_json")
    fun getindex2(param: Param): Data {
        val customerList = customerService!!.customerList
        val alias = ArrayList<String>()
        alias.add("zk")
        alias.add("zk2")
        customerList[0].alias = alias
        return Data(customerList)
    }

    @Action("get:/customer_create")
    fun create(param: Param): View {
        return View("customer_edit.jsp")
    }

    @Action("post:/customer_create")
    fun createSubmit(param: Param): Data {
        val fields = param.map
        val customer = customerService!!.createCustomer(fields)
        return Data(customer)
    }

    @Action("get:/customer_edit")
    fun edit(param: Param): View {
        val id = param.getLong("id")
        val customer = customerService!!.getCustomer(id)
        return View("customer_edit.jsp").addModel("customer", customer)
    }

    @Action("put:/customer_edit")
    fun editSubmit(param: Param): Data {
        val id = param.getLong("id")
        val fields = param.map
        val customer = customerService!!.updateCustomer(id, fields)
        return Data(customer)
    }

    @Action("get:/customer_delete")
    fun delete(param: Param): View {
        val id = param.getLong("id")
        val aBoolean = customerService!!.deleteCustomer(id)

        val customerList = customerService!!.customerList
        return View("customer.jsp").addModel("customerList", customerList)
    }
}
